### ISSUE TITLE
Source mapping points to incorrect path (still/again)

### ISSUE THREAD

**Issue by lehnerpat** [mmomtchev/rlayers#37]
Hi, thanks so much for publishing this package, it really helps a lot with using openlayers in React!

When I use rlayers (v1.3.1) with create-react-app, I get a lot of warnings like:
```
WARNING in ./node_modules/rlayers/control/RAttribution.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '$PROJECT_DIR/node_modules/rlayers/control/src/control/RAttribution.tsx' file: Error: ENOENT: no such file or directory, open '$PROJECT_DIR/node_modules/rlayers/control/src/control/RAttribution.tsx'
 @ ./node_modules/rlayers/control/index.js 23:21-46
 @ ./node_modules/rlayers/index.js 249:32-52
 @ ./src/OlTest.tsx 6:0-71 16:35-39 23:38-42 27:37-49 29:39-47 35:41-49
 @ ./src/App.tsx 13:0-34 34:41-47
 @ ./src/index.tsx 7:0-24 11:33-36
``` 

The warning is correct: the .tsx files aren't located in `node_modules/rlayers/control/src/control`, they're in `node_modules/rlayers/src/control` (the source mapping references `src/` but should instead reference `../src/` when it's in a subdirectory.

It would be great if you could fix this.

(This is related to #25 but seems to be a new issue resulting from its fix.)

**Comment by lehnerpat** [comment]
Update: from trying around a bit myself, the quickest fix is changing tsc's `sourceRoot` and `outDir` to its default (which is right next to each .ts file), and then just packaging the whole workspace.

tsconfig.json change:
```diff
diff --git a/tsconfig.json b/tsconfig.json
index 0250660..c782d8c 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,18 +1,16 @@
 {
     "compilerOptions": {
         "sourceMap": true,
-        "sourceRoot": "src",
         "noImplicitAny": false,
         "module": "commonjs",
         "target": "es5",
         "lib": [
             "esnext",
             "dom",
             "dom.iterable"
         ],
-        "outDir": "./dist",
         "declaration": true,
         "declarationMap": true,
         "removeComments": true,
         "allowSyntheticDefaultImports": true,
         "jsx": "react",
```

This is annoying during development because it clutters your workspace, so some workarounds are probably desirable.

**Comment by lehnerpat** [comment]
I worked out a temporary work-around for users of rlayers! 
I'm sharing this here **as-is, without any guarantees or warranties about its correctness. USE AT YOUR OWN RISK.**

Put the script below into the root of your project as an .mjs file (e.g. `fix-rlayers.mjs`), and install the npm packages `fs-extra` and `globby` if you don't already have them.
Then run `node fix-rlayers.mjs` from the root of your project.

You can optionally add a script entry to your `package.json` if you want, maybe even the [`postinstall` script](https://docs.npmjs.com/cli/v6/using-npm/scripts#npm-install), which is run automatically after you run `npm install`; this should help the fix be applied even in a CI environment.

`fix-rlayers.mjs`:
```js
import fs from "fs-extra";
import { globbySync as globby } from "globby";

process.chdir("node_modules/rlayers");

fs.copySync("src", ".", { recursive: true });
const allMaps = globby("**/*.map");
for (const mapPath of allMaps) {
  const m = fs.readJSONSync(mapPath);
  m["sourceRoot"] = "";
  m["sources"] = m["sources"].map((s) => {
    const parts = s.split("/");
    return parts[parts.length - 1];
  });
  fs.writeJSONSync(mapPath, m);
}
```

**Comment by mmomtchev** [comment]
You seem to have some 3rd party source map loader installed - it is this loader that generates the message.

I will probably end up completely removing the source maps, as there seem to be various extensions that can read them and given that they are used only for debugging, they are probably of no interest to most of the users.

**Comment by fogracvxy** [comment]
> You seem to have some 3rd party source map loader installed - it is this loader that generates the message.
> 
> I will probably end up completely removing the source maps, as there seem to be various extensions that can read them and given that they are used only for debugging, they are probably of no interest to most of the users.


Fixed it using @lehnerpat fix .mjs file.

Keep up the good work!

**Comment by mmomtchev** [comment]
Can you please test if your tools work with this source map in `control/RFullScreen.js.map`

```json
{"version":3,"file":"RFullScreen.js","sourceRoot":"rlayers/src/","sources":["control/RFullScreen.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AACA,sCAAsC;AAItC,gEAAsE;AAoBtE;IAAyC,+BAAqD;IAG1F,qBAAY,KAAiC,EAAE,OAAoC;QAAnF,YACI,kBAAM,KAAK,EAAE,OAAO,CAAC,SAExB;QADG,KAAI,CAAC,EAAE,GAAG,IAAI,oBAAU,CAAC,KAAI,CAAC,SAAS,CAAC,KAAK,CAAC,CAAC,CAAC;;IACpD,CAAC;IAED,+BAAS,GAAT,UAAU,KAAuB;QAC7B,6BACO,iBAAM,SAAS,YAAC,KAAK,CAAC,KACzB,MAAM,EAAE,KAAK,CAAC,MAAM,EACpB,KAAK,EAAE,KAAK,CAAC,KAAK,EAClB,WAAW,EAAE,KAAK,CAAC,WAAW,EAC9B,QAAQ,EAAE,KAAK,CAAC,QAAQ,EACxB,SAAS,EAAE,KAAK,CAAC,SAAS,IAC5B;IACN,CAAC;IACL,kBAAC;AAAD,CAAC,AAlBD,CAAyC,sBAAY,GAkBpD"}
```

**Comment by mmomtchev** [comment]
Should be fixed in 1.3.2
